lib/game.rb

```diff
@@ -0,0 +1,90 @@
+class Game
+  attr_reader :rolls
+
+  MAX_PINS = 10
+
+  def initialize
+    @rolls = [] 
+    @max_rolls = 20
+  end
+
+  def roll(pins) 
+    check_pin_number
+    check_game_status
+    @rolls << pins
+      if @rolls.count == 19 && (pins == 10) || @rolls.count == 20 && (@rolls[-1] + @rolls[-2]) == 10
+        @max_rolls += 1
+      elsif first_bowl? && (pins == 10)
+        @rolls << "-" unless completed_all_rolls
+      end
+  end
+
+  def check_game_status
```

Comment:
This guard clause should probably be private. Also, the name is not super explicit, maybe "check_for_completed_game" or something like that?